### PR TITLE
Honor features and expose potential support to `CodeGenerator`

### PR DESCRIPTION
**Pull request by thomasvl**
This does not enable editions for Swift Protobuf, this is just more internals to build things out.

### CODE REVIEW

File: Sources/SwiftProtobufPluginLibrary/CodeGenerator.swift [hunk 1]
```diff
@@ -139,6 +144,11 @@ extension CodeGenerator {
 // Provide default implementation for things so `CodeGenerator`s only have to
 // provide them if they wish too.
 extension CodeGenerator {
+  public var supportedEditionRange: ClosedRange<Google_Protobuf_Edition> {
+    // Default impl of unknow so generator don't have to provide this until
```
Comment (allevato) [inline Sources/SwiftProtobufPluginLibrary/CodeGenerator.swift:148]:
```suggestion
    // Default impl of unknown so generator don't have to provide this until
```

File: Sources/SwiftProtobufPluginLibrary/Descriptor.swift [hunk 2]
```diff
@@ -73,6 +84,10 @@ public final class DescriptorSet {
     featureSetDefaults: Google_Protobuf_FeatureSetDefaults,
     featureExtensions: [any AnyMessageExtension] = []
   ) {
+    precondition(Self.bundledEditionsSupport.contains(featureSetDefaults.minimumEdition),
+                 "Attempt to use a FeatureSetDefault miniumEdition that isn't supported by the library.")
```
Comment (allevato) [inline Sources/SwiftProtobufPluginLibrary/Descriptor.swift:88]:
```suggestion
                 "Attempt to use a FeatureSetDefault minimumEdition that isn't supported by the library.")
```

**Review by allevato — APPROVED** [review]

File: Sources/SwiftProtobufPluginLibrary/CodeGenerator.swift (hunk 1 above)
Comment (thomasvl) [reply Sources/SwiftProtobufPluginLibrary/CodeGenerator.swift]: Done.

File: Sources/SwiftProtobufPluginLibrary/Descriptor.swift (hunk 2 above)
Comment (thomasvl) [reply Sources/SwiftProtobufPluginLibrary/Descriptor.swift]: Done

**Pull request merged**